validator-core/src/main/java/fr/ign/validator/database/internal/ConditionMismatchFinder.java

```diff
@@ -0,0 +1,68 @@
+package fr.ign.validator.database.internal;
+
+import java.io.IOException;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+import fr.ign.validator.database.Database;
+import fr.ign.validator.database.RowIterator;
+
+/**
+ * Condition Mismatch Finder - query the database to get all the features who
+ * did not respect FeatureType conditions
+ * 
+ * @author cbouche
+ *
+ */
+public class ConditionMismatchFinder {
+
+    /**
+     * hardcoded limit to avoid huge report on massively invalid table.
+     */
+    public static final int LIMIT_ERROR_COUNT = 10;
+
+    /**
+     * A precise reference to the id and the file where the constraint failed
+     * 
+     * @author cbouche
+     */
+    public class ConditionMismatch {
+        public String id;
+        public String file;
+
+        public ConditionMismatch(String id, String file) {
+            this.id = id;
+            this.file = file;
+        }
+    }
+
+    /**
+     * 
+     * @param database
+     * @param tableName
+     * @param condition
+     * @return
+     * @throws SQLException
+     * @throws IOException
+     */
+    public List<ConditionMismatch> findConditionMismatch(Database database, String tableName, String condition)
+        throws SQLException, IOException {
+
+        String query = "SELECT __id, __file "
+            + " FROM " + tableName
+            + " WHERE NOT (" + condition + ")"
```

Comment:
Here is the dangerous part. Add a TODO to validate "condition" string to allow the use of "conditions" on use provided models?